packages/engine-core/src/data-proxy/utils/request.ts

```diff
@@ -1,34 +1,40 @@
 import type { IncomingMessage } from 'http'
-import type _https from 'https'
+import type Https from 'https'
+import type { RequestInit, Response } from 'node-fetch'
 import type { O } from 'ts-toolbelt'
 
+import { RequestError } from '../errors/NetworkError'
 import { getJSRuntimeName } from './getJSRuntimeName'
 
 // our implementation handles less
-export type RequestOptions = O.Patch<
-  {
-    headers?: { [k: string]: string }
-    body?: string
-  },
-  RequestInit
->
-
-// our implementation handles less
+export type RequestOptions = O.Patch<{ headers?: { [k: string]: string }; body?: string }, RequestInit>
```

Comment:
Note: it seems this change prevents TypeScript to properly build `engine-core`, as it affects the `buildOptions` function.

Temporary solution:

```typescript
function buildOptions(options: RequestOptions): Https.RequestOptions {
  return {
    // @ts-ignore
    method: options.method,
    headers: buildHeaders(options),
  }
}
```